src/main/java/io/temporal/internal/replay/WorkflowExecutor.java

```diff
@@ -24,45 +24,53 @@
 import io.temporal.api.query.v1.WorkflowQuery;
 import io.temporal.api.query.v1.WorkflowQueryResult;
 import io.temporal.api.workflowservice.v1.PollWorkflowTaskQueueResponseOrBuilder;
+import java.time.Duration;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
 public interface WorkflowExecutor {
 
-  WorkflowTaskResult handleWorkflowTask(PollWorkflowTaskQueueResponseOrBuilder workflowTask)
-      throws Throwable;
+  /**
+   * Handles a single workflow task.
+   *
+   * @param workflowTask task to handle
+   * @return true if new task should be created synchronously as local activities are still running.
+   */
+  WorkflowTaskResult handleWorkflowTask(PollWorkflowTaskQueueResponseOrBuilder workflowTask);
 
   Optional<Payloads> handleQueryWorkflowTask(
-      PollWorkflowTaskQueueResponseOrBuilder workflowTask, WorkflowQuery query) throws Throwable;
+      PollWorkflowTaskQueueResponseOrBuilder workflowTask, WorkflowQuery query);
 
   void close();
 
+  Duration getWorkflowTaskTimeout();
+
   class WorkflowTaskResult {
+
     private final List<Command> commands;
-    private final boolean forceCreateNewWorkflowTask;
     private final boolean finalCommand;
     private final Map<String, WorkflowQueryResult> queryResults;
+    private final boolean forceWorkflowTask;
 
     public WorkflowTaskResult(
         List<Command> commands,
         Map<String, WorkflowQueryResult> queryResults,
-        boolean forceCreateNewWorkflowTask,
-        boolean finalCommand) {
+        boolean finalCommand,
```

Comment:
we swapped the order to two arguments with the same type...hopefully all the callers have been updated

Reply:
Agree that it is error-prone. [Changed](https://github.com/temporalio/java-sdk/pull/172/commits/68db6c6c2da1c4fce870ff2c41cbfdade02928cd) WorkflowTaskResult to use the Builder pattern.